docs/DeveloperGuide.md

```diff
@@ -666,152 +666,351 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**Preconditions:** User has the SOCareers App open.  
+**Guarantees:** A link to the user guide is provided to the user.  
+**MSS:**
+1. User requests for help.
+2. SOCareers App displays a pop-up box with a link to the user guide.
+
+    Use case ends.
+
+---
+
+#### Use Case: UC02 - Adding a Person
+
+**Preconditions:** User has the SOCareers App open.
+**Guarantees:** A new person is added to the user’s list of people with the provided details.  
+**MSS:**
+1. User requests to add a person contact and provides the person's details.
+2. SOCareers adds the person to the list of people and displays confirmation in the UI.
+
+    Use case ends.
+
+**Extensions:**
+- 1a. The provided information is invalid (e.g., invalid phone number format, invalid email format, duplicate contact, etc.).
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+
+
+---
+
+#### Use Case: UC03 - Adding a Company
+
+**Preconditions:** User has the SOCareers App open.
+**Guarantees:** A new company is added to the user’s list of companies with the provided details.  
+**MSS:**
+1. User requests to add a company contact and provides the company's details.
+2. SOCareers adds the company to the list of companies and displays confirmation in the UI.
+
+    Use case ends.
+
+**Extensions:**
+- 1a. The provided information is invalid (e.g., invalid phone number format, invalid email format, duplicate contact, etc.).
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+
+
+---
+
+#### Use Case: UC04 - Adding an Internship to a Company Contact
+
+**Preconditions:** User has the SOCareers App open and a list of companies available.
+**Guarantees:** A new internship is added to the specified company’s internships list with the provided details.  
+**MSS:**
+1. User requests to add an internship to a company contact and provides the internship's details.
+2. SOCareers adds the internship to the specified company's list of internships and displays confirmation in the UI.
+
+    Use case ends.
 
-1. User requests to list companies.
-2. SOCareers shows a list of companies.
-3. User requests to add a new company to the list.
-4. SOCareers prompts the user to enter the company details.
-5. User enters the company details.
-6. SOCareers validates the provided details.
-7. SOCareers adds the company to the list.
-8. SOCareers displays a message confirming the addition of the company.
+**Extensions:**
+- 1a. The provided information is invalid (e.g., invalid company to add to, invalid datetime, etc.).
+  - 1a1. SOCareers displays an error message.
 
-   Use case ends.
+    Use case resumes from step 1.
 
-**Extensions**
 
-* 2a. The list is empty.
+---
 
-  Use case ends.
+#### Use Case: UC05 - Listing All Persons
 
-* 6a. SOCareers detects an error in the entered details.
+**Preconditions:** User has the SOCareers App open and has added at least one person to their list of people.
```

Comment:
I think you can leave out the added since the app does come with sample data? We can consider rephrasing to 'has at least one person in their list of people'